cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -309,6 +323,43 @@ public void run() {
       }
     }
 
+    private Function<Object, Object> attemptToTransformCredentials = new Function<Object, Object>() {
+      @Nullable
+      @Override
+      public Object apply(@Nullable Object input) {
+        input = attemptExtractX509Certificate(input);
+        return attemptExtractPrivateKey(input);
+      }
+    };
+
+    private Object attemptExtractX509Certificate(Object value) {
+      if (value != null && value instanceof String && value.toString().contains("BEGIN CERTIFICATE")) {
+        try {
+          X509CertParser x509CertParser = new X509CertParser();
+          x509CertParser.engineInit(new ByteArrayInputStream(value.toString().getBytes()));
+          return x509CertParser.engineRead();
+        } catch (Exception ex) {
```

Comment:
ok to catch Exception if catching the actual possible exception types is arduous.